DpcWaitOnProcess/Main.cpp

```diff
@@ -122,38 +114,20 @@ DriverUnload(
     if (g_DpcWait != nullptr)
```

Comment:
I think this check is race-free, because after you unregister notification routines, any pending callbacks must exist first, and no new callbacks are allowed -- so the race is only in your callback path.